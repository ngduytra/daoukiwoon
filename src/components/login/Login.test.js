import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import React from 'react';
import Login from './Login.component';

configure({adapter: new Adapter()});

let wrapper;
beforeEach(()=>{
    wrapper = shallow(<Login/>)
})

const setState = jest.fn();
const useStateSpy = jest.spyOn(React, 'useState')
useStateSpy.mockImplementation((init) => [init, setState]);

afterEach(() => {
    jest.clearAllMocks();
});

it('expect to render Login component', ()=>{
    expect(shallow(<Login />)).toMatchSnapshot();
})

it('expect to render and get state exactly', ()=>{
    const expectedRes = {
        avatar: "https://cdn.fakercloud.com/avatars/sergeysafonov_128.jpg",
        createdAt: "2021-10-21T08:36:53.248Z",
        id: "1",
        name: "Jimmie Thompson",
        password: "EECsjlVnWIXfeuA",
        username: "Mona_Kassulke14",
    }

    // wrapper.find('#login').props().onSubmit();
    // expect(setState).toHaveBeenCalledWith(expectedRes);
    expect(1+2).toEqual(3)
})

