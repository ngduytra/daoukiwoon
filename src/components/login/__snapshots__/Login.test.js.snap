// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Login component 1`] = `ShallowWrapper {}`;
